x-pack/plugins/fleet/server/services/package_policy.ts

```diff
@@ -938,62 +927,57 @@ export function overridePackageInputs(
             ),
             package: { name: packageName, version: basePackagePolicy.package!.version },
           };
+
           if (dryRun) {
             errors.push({
               key: `${override.type}.streams.${streamSet}`,
               message: String(e.error),
             });
+
             continue;
-          } else throw e;
+          } else {
+            throw e;
+          }
         }
 
-        if (typeof stream.enabled !== 'undefined') originalStream.enabled = stream.enabled;
+        if (typeof stream.enabled !== 'undefined') {
+          originalStream.enabled = stream.enabled;
+        }
 
         if (stream.vars) {
-          try {
-            const { result, missingVars } = deepMergeVars(stream as InputsOverride, originalStream);
-            originalStream = result;
-            responseMissingVars = [...responseMissingVars, ...missingVars];
-          } catch (e) {
-            const varName = e.message;
-            const streamSet = stream.data_stream.dataset;
-            const err = {
-              error: new Error(
-                i18n.translate('xpack.fleet.packagePolicyStreamVarOverrideError', {
-                  defaultMessage:
-                    'Var {varName} does not exist on {streamSet} for {inputType} of package {packageName}',
-                  values: {
-                    varName,
-                    streamSet,
-                    inputType: override.type,
-                    packageName,
-                  },
-                })
-              ),
-              package: { name: packageName, version: basePackagePolicy.package!.version },
-            };
-            if (dryRun) {
-              errors.push({
-                key: `${override.type}.streams.${streamSet}.${varName}`,
-                message: String(err.error),
-              });
-            } else throw err;
-          }
+          originalStream = deepMergeVars(originalStream, stream as InputsOverride);
         }
       }
     }
   }
 
+  const resultingPackagePolicy: NewPackagePolicy = {
+    ...basePackagePolicy,
+    inputs,
+  };
+
+  const validationResults = validatePackagePolicy(resultingPackagePolicy, packageInfo);
```

Comment:
Really cool to see this coming server side, we should probably put it in the create|update package policy too (not in this PR)

Reply:
Created an issue for that https://github.com/elastic/kibana/issues/108927